core/dogu_v2.go

```diff
@@ -0,0 +1,490 @@
+func (d *Dogu) GetEnvironmentVariables() []EnvironmentVariable {
+	return d.EnvironmentVariables
+}
+
+// GetEnvironmentVariablesAsStringArray returns environment variables formatted as string array
+func (d *Dogu) GetEnvironmentVariablesAsStringArray() []string {
+	var environmentVariables []string
+	for _, environmentVariable := range d.EnvironmentVariables {
+		environmentVariables = append(environmentVariables, environmentVariable.String())
+	}
+	return environmentVariables
+}
+
+// DependsOn returns true if the dogu has a hard dependency to the given dogu
+func (d *Dogu) DependsOn(name string) bool {
+	dependencies := d.GetDependenciesOfType(DependencyTypeDogu)
+	if dependencies == nil {
+		return false
+	}
+
+	for _, dependency := range dependencies {
+		if dependency.Name == name {
+			return true
+		}
+	}
+
+	return false
+}
+
+// GetVersion parses the dogu version and returns a struct which can be used to compare versions
+func (d *Dogu) GetVersion() (Version, error) {
+	version, err := ParseVersion(d.Version)
+	if err != nil {
+		return version, fmt.Errorf("failed to parse version %s of dogu %s: %w", d.Version, d.Name, err)
+	}
+	return version, nil
+}
+
+// IsEqualTo returns true if the other dogu has the same name and version.
+func (d *Dogu) IsEqualTo(otherDogu *Dogu) (bool, error) {
+	if d.Name != otherDogu.Name {
+		return false, fmt.Errorf("only dogus with the same name can be compared")
+	}
+
+	version, err := d.GetVersion()
+	if err != nil {
+		return false, err
+	}
+
+	otherVersion, err := otherDogu.GetVersion()
+	if err != nil {
+		return false, err
+	}
+
+	return version.IsEqualTo(otherVersion), nil
+}
+
+// IsNewerThan returns true if the other dogu has the same name and has a higher version
+func (d *Dogu) IsNewerThan(otherDogu *Dogu) (bool, error) {
+	if d.Name != otherDogu.Name {
+		return false, fmt.Errorf("only dogus with the same name can be compared")
+	}
+
+	version, err := d.GetVersion()
+	if err != nil {
+		return false, err
+	}
+
+	otherVersion, err := otherDogu.GetVersion()
+	if err != nil {
+		return false, err
+	}
+
+	return version.IsNewerThan(otherVersion), nil
+}
+
+// PrintDogus prints out dogu name, version, display name and description in a table
+func PrintDogus(dogus []*Dogu, ns bool) {
+	table := uitable.New()
+	table.Separator = "  "
+	table.MaxColWidth = 50
+
+	table.AddRow("")
+	table.AddRow("NAME", "VERSION", "DISPLAYNAME", "DESCRIPTION")
+	for _, dogu := range dogus {
+		var name string
+		if ns {
+			name = dogu.GetFullName()
+		} else {
+			name = dogu.GetSimpleName()
+		}
+		table.AddRow(name, dogu.Version, dogu.DisplayName, dogu.Description)
+	}
+
+	GetLogger().Println(table)
+}
+
+func GetSimpleDoguName(fullDoguName string) string {
+	dogu := Dogu{Name: fullDoguName}
+	return dogu.GetSimpleName()
+}
+
+func GetNamespace(fullDoguName string) string {
+	dogu := Dogu{Name: fullDoguName}
+	return dogu.GetNamespace()
+}
+
```

Comment:
Here, too, the code lacks doc comments for public symbols.
```suggestion
// CreateV1Copy converts this dogu object into a deep-copied DoguV1 object (for legacy reasons).
```
- [ ]

Reply:
Done